### PR TITLE
First take at creating a GitHub issue template

### PR DESCRIPTION
**Pull request by hamoid**

### CODE REVIEW

**Comment by arturoc** [comment]
That's really nice, i think there's a way to show this as a message for people sending issues for the first time, i'll take a look when i have a moment

**Comment by hamoid** [comment]
https://github.com/blog/2111-issue-and-pull-request-templates

**Comment by hamoid** [comment]
It's also possible to create a template for pull requests, but I have no experience with what's a good template. Anything you want people to write in a pull request? Some kind of reminder?

**Comment by bakercp** [comment]
Hey there, I just took a first pass at some edits additions.  The main goal of my additions was to differentiate between "questions" (which go to forum) and "issues" which are documented and (hopefully) reproducible.

**Comment by hamoid** [comment]
Hi @bakercp ! where are your edits visible?

**Comment by bakercp** [comment]
I just put them in the review comments ...

**Comment by hamoid** [comment]
I don't know where to click to see the comments. Are they only visible to members of the OF organization?

**Comment by bakercp** [comment]
Oh.  Apparently you have to press "Submit Review" ... can you see it now?

**Comment by hamoid** [comment]
I see it now, thanks! I'll add my comments soon.

**Comment by bakercp** [comment]
Hey, what do you think of this check-box approach?  Too much?

https://github.com/pi-hole/pi-hole/blob/master/.github/ISSUE_TEMPLATE.md

**Comment by hamoid** [comment]
That one in particular reminds me a bit of government forms, a bit too serious :) But the checkboxes may be useful. 

I'll first push some changes following your comments, and lets continue from there.

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+Nice! You found an issue! :) Get help faster by providing the following details:
```
Comment (hamoid) [inline .github/ISSUE_TEMPLATE.md]: About "code to demonstrate it", what if it's an improvement or new feature? Isn't it enough to have the block of code at the end to imply this?

**Comment by hamoid** [comment]
Another topic: should it be mention that it's not just issues vs questions, but issues in the openFrameworks code vs issues in your own code?

**Comment by hamoid** [comment]
Some template examples: https://github.com/stevemao/github-issue-templates

**Comment by arturoc** [comment]
This is looking great but isn't it a bit too long if it's what everyone is going to see when submitting an issue? i don't think the explanation of issue vs question belongs here so tbh i prefer your first version or  something more concise even.

@bakercp what do you think?

**Comment by bakercp** [comment]
Minimal is fine.  I agree, it's a lot of text in its current form, so I'd say we should refactor some of the more instructive / pedagogical bits into a refined contributing document.  My main goals are to get a good report the first time, rather than having long back-and-forth conversations trying extract information from people.  So the basics would just include ...

- Help people figure out if they should post here or on the forum (question vs. issue).
-- This could be a single sentence with a link more text / examples in the contributing readme.
- Get info on hardware / software versions.
- Get expected vs. actual results (with images if relevant)
- Get a minimal example / steps to reproduce (with code fences!)

**Comment by arturoc** [comment]
I don't think the explanation of question/issue belongs in a template really, this is something that will appear for everyone when filling an issue so it should be really straightforward to just fill in. And there it shouldn't get in the way, you shouldn't have to remove anything except perhaps to replace it i think.

Having to delete that bit even if it's just one sentence every time you open an issue can get annoying i think

Also if people see the rest of the template (hardware, description of the issue, minimal example...) i think most people will understand that it's referring to an issue and not a question

I really prefer to point to the forum and close a couple of questions every year (i don't think the frequency right now is much higher than that) than having to delete parts of the template every time i open an issue

**Comment by hamoid** [comment]
I also feel it's too much information to read. We could add a simple "If this is your first issue, please read this before continuing" at the top. 

This way we can keep it short, and put the details and explanations in the wiki. Would that work better?

**Comment by bakercp** [comment]
That sounds like the right approach.

**Comment by hamoid** [comment]
Should I add **Getting help with OpenFrameworks** at the top of the [wiki](https://github.com/openframeworks/openFrameworks/wiki) ? Alternatives names:

* Asking for help
* Asking for help: forum vs issues
* How to ask for help
* I need help with OpenFrameworks
* ... 

I'll move text out from `.github/ISSUE_TEMPLATE.md` into that wiki page, which will explain when to use the forum and when to post issues, and how to post a useful issue.

At the top of the template I'll write: 
> Read https://github.com/openframeworks/openFrameworks/wiki/asking-for-help before posting your first OpenFrameworks issue

How does that sound like?

**Comment by hamoid** [comment]
Now it's possible to create multiple templates for different kinds of issues:
https://blog.github.com/2018-05-02-issue-template-improvements/

**Comment by dimitre** [comment]
My favorite template is the blank one, but maybe a really basic template would have
```
platform and OS version:
OF version:
```

**Pull request closed without merging**